client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/TraceContext.java

```diff
@@ -133,11 +130,13 @@ public void endProcessingNode() {
                 recorder.report(trace);
             }
 
-            if (!rootNode.hasProperty(Constants.PROP_SERVICE_NAME)) {
-                rootNode.addProperty(this.serviceName);
-            }
-            if (!rootNode.hasProperty(Constants.PROP_BUILD_STAMP)) {
-                rootNode.addProperty(this.buildTimeStamp);
+            if (!(this.deploymentMetaData == null)) {
```

Comment:
As mentioned above, think this logic would be better inside the EnvAwareTraceRecorder.